# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../fribidi/PAGE_SIZE.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- fribidi-0.19.1/lib/common.h.vanilla	2008-12-07 19:19:56.000000000 +0100
+++ fribidi-0.19.1/lib/common.h	2008-12-07 19:29:13.000000000 +0100
@@ -38,6 +38,8 @@
 
 #include <fribidi-common.h>
 
+#include <unistd.h>
+
 /* FRIBIDI_PRIVATESPACE is a macro used to name library internal symbols. */
 #ifndef FRIBIDI_PRIVATESPACE
 # define FRIBIDI_PRIVATESPACE(SYMBOL) FRIBIDI_NAMESPACE(_##SYMBOL##__internal__)
@@ -136,7 +138,7 @@
 #  ifndef __FRIBIDI_DOC
 #   include <asm/page.h>
 #  endif /* __FRIBIDI_DOC */
-#  define FRIBIDI_CHUNK_SIZE (PAGE_SIZE - 16)
+#  define FRIBIDI_CHUNK_SIZE (sysconf(_SC_PAGESIZE) - 16)
 # else /* !HAVE_ASM_PAGE_H */
 #  define FRIBIDI_CHUNK_SIZE (4096 - 16)
 # endif	/* !HAVE_ASM_PAGE_H */
